app/src/main/java/org/gdg/frisbee/android/cache/ModelCache.java

```diff
@@ -360,42 +358,35 @@ private long readExpirationFromDisk(InputStream is) throws IOException {
         return expiration;
     }
 
-    private Object readValueFromDisk(InputStream is) throws IOException {
+    private Object readValueFromDisk(InputStream is) throws IOException, ClassNotFoundException {
         BufferedReader fss = new BufferedReader(new InputStreamReader(is, "UTF-8"));
         String className = fss.readLine();
 
         if (className == null) {
             return null;
         }
 
-        String line = null;
+        String line;
         String content = "";
         while ((line = fss.readLine()) != null) {
             content += line;
         }
 
         fss.close();
 
-        Class<?> clazz;
-        try {
-            if (className.contains("google")) {
-                clazz = Class.forName(className);
-                return mJsonFactory.createJsonParser(content).parseAndClose(clazz, null);
-            } else {
-                clazz = Class.forName(className);
-                return mGson.fromJson(content, clazz);
-            }
-        } catch (IllegalArgumentException e) {
-            Timber.e(e, "Deserializing from disk failed");
-            return null;
-        } catch (ClassNotFoundException e) {
-            throw new IOException(e.getMessage());
+        if (className.contains("google")) {
```

Comment:
Good to remove the try catch! But could do with some explanations :-) i.e. that it can throw an IllegalArgumentException